app/models/user/reputation_token.rb

```diff
@@ -32,8 +32,9 @@ class User::ReputationToken < ApplicationRecord
     self.value = self.determine_value
   end
 
-  after_save_commit do
-    summing_sql = Arel.sql("(#{user.reputation_tokens.select('SUM(value)').to_sql})")
+  after_commit do
```

Comment:
This needed to change to `after_commit` as otherwise the rep was not recalculated after deleting the token.